### ISSUE TITLE
Migrate to LangChainJS

### ISSUE THREAD

**Issue by logancyang** [logancyang/obsidian-copilot#16]
This is a significant change to this plugin and will make a lot more features possible, such as better conversation memory, unlimited context, augmented retrieval, and much more. LangChainJS has many existing features for the functionalities I already implemented in this plugin as well.

Will bump major version after this migration.

**Comment by logancyang** [comment]
Currently facing an issue with esbuild and wasm

```
[watch] build finished, watching for changes...
✘ [ERROR] No loader is configured for ".wasm" files: node_modules/@dqbd/tiktoken/tiktoken_bg.wasm

    node_modules/@dqbd/tiktoken/tiktoken.js:1:22:
      1 │ import * as wasm from "./tiktoken_bg.wasm";
```

Raised here: https://github.com/hwchase17/langchainjs/issues/1276

**Comment by logancyang** [comment]
> Currently facing an issue with esbuild and wasm
> 
> ```
> [watch] build finished, watching for changes...
> ✘ [ERROR] No loader is configured for ".wasm" files: node_modules/@dqbd/tiktoken/tiktoken_bg.wasm
> 
>     node_modules/@dqbd/tiktoken/tiktoken.js:1:22:
>       1 │ import * as wasm from "./tiktoken_bg.wasm";
> ```
> 
> Raised here: [hwchase17/langchainjs#1276](https://github.com/hwchase17/langchainjs/issues/1276)

Resolved by adding a wasm plugin for esbuild. LangChainJS works in this plugin now.

**Comment by logancyang** [comment]
Published 2.0.0 after merging the langchain integration change.
